### PR TITLE
Copy go.mod into the build dir

### PR DESCRIPTION
**Pull request by gagliardetto**
This PR contents:

- When pretty-printing the generated source code, use `golang.org/x/tools/imports` which also fixes imports before formatting the source code.
- Copy the `go.mod` file (if available) into the temporary build folder.
- Update `golang.org/x/tools` (needed to have `golang.org/x/tools/imports`).

### CODE REVIEW

**Comment by gagliardetto** [comment]
I'll probably better skip the vendoring part.

**Comment by gagliardetto** [comment]
Ok. I removed the /vendor/ files from this PR.

But in order for depstubber to compile, the vendoring needs to be re-done with the updated go.mod.

**Comment by gagliardetto** [comment]
Please run `go mod vendor` so you don't have to review that code.

File: reflect.go [hunk 1]
```diff
@@ -120,8 +121,54 @@ func runInDir(program []byte, dir string) (*model.PackedPkg, error) {
...
+		if err != nil {
+			log.Fatalf("Unable to load current directory: %v", err)
+		}
+
+		modRoot := findModuleRoot(wd)
+
+		if modRoot != "" {
+			MustCopyFile(filepath.Join(modRoot, "go.mod"), filepath.Join(tmpDir, "go.mod"))
+		}
+
```
Comment (sauyon) [inline reflect.go:136]: Could you remove this newline? Is this autoformatted?

File: reflect.go [hunk 2]
```diff
@@ -120,8 +121,54 @@ func runInDir(program []byte, dir string) (*model.PackedPkg, error) {
...
 	return run(filepath.Join(tmpDir, progBinary))
 }
+func MustCopyFile(src, dst string) {
+	_, err := copyFile(src, dst)
+	if err != nil {
+		log.Fatalf("error copying %q to %q: %s", src, dst, err)
+	}
+}
+
+func copyFile(src, dst string) (int64, error) {
```
Comment (sauyon) [inline reflect.go:148]: Why not just use `io.Copy`?

**Review by sauyon — COMMENTED** [review]
Please run `go mod vendor` and just put that in a separate commit, that should be fine to review.

Why are you switching to using `imports`? Were you running into issues with imports before?

File: reflect.go (hunk 1 above)
Comment (gagliardetto) [reply reflect.go]: > Could you remove this newline?

Removed.

> Is this autoformatted?

Yes, it is. The standard Go formatter is completely fine with newlines in positions like that. For a stricter formatter, see https://github.com/mvdan/gofumpt

File: reflect.go (hunk 2 above)
Comment (gagliardetto) [reply reflect.go:147]: `copyFile` does use `io.Copy` under the hood.

`io.Copy` does not allow to directly do the whole file creation and copy: here's the signature of `io.Copy`: 

`func Copy(dst Writer, src Reader) (written int64, err error)`

**Comment by gagliardetto** [comment]
> Why are you switching to using imports? Were you running into issues with imports before?

Sometimes there are unused imports in the generated files; that does not allow them to compile.

The `golang.org/x/tools/imports` package does use `format.Source` after fixing the imports:

https://github.com/golang/tools/blob/72fbef54948b/imports/forward.go#L66

https://github.com/golang/tools/blob/72fbef54948b223c8a7e1fdb1f23c0d80442e33d/internal/imports/imports.go#L60

https://github.com/golang/tools/blob/72fbef54948b223c8a7e1fdb1f23c0d80442e33d/internal/imports/imports.go#L150

File: reflect.go (hunk 2 above)
Comment (sauyon) [reply reflect.go:147]: :facepalm: yes, I failed to read.

File: reflect.go (hunk 1 above)
Comment (sauyon) [reply reflect.go]: As long as it's autoformatted I'm not particularly fussed.

**Review by sauyon — APPROVED** [review]
Ah, unused imports in the stubbed file, I see. LGTM, thanks!

**Pull request merged**